File: pytools/data/data.code.php.client/ReqRoleCreate.php
<?php
namespace protocol;

use network\Packet;


class ReqRoleCreate
{
	private $uid = 0;
	private $uuid = 0;
	private $sid = 0;
	private $cid = 0;
	private $os = '';
	private $version = '';
	private $uname = '';
	private $source = '';
	private $sourceSub = '';
	private $loginTime = 0;


	public function encode()
	{
		$packet = new Packet();
		$packet->writeU32($this->uid);
		$packet->writeU32($this->uuid);
		$packet->writeU16($this->sid);
		$packet->writeU16($this->cid);
		$packet->writeString($this->os);
		$packet->writeString($this->version);
		$packet->writeString($this->uname);
		$packet->writeString($this->source);
		$packet->writeString($this->sourceSub);
		$packet->writeU32($this->loginTime);
		return $packet->encode(Msg::$P_REQ_ROLE_CREATE);
	}


	public function setUid($uid)
	{
		$this->uid = $uid;
	}

	public function setUuid($uuid)
	{
		$this->uuid = $uuid;
	}

	public function setSid($sid)
	{
		$this->sid = $sid;
	}

	public function setCid($cid)
	{
		$this->cid = $cid;
	}

	public function setOs($os)
	{
		$this->os = $os;
	}

	public function setVersion($version)
	{
		$this->version = $version;
	}

	public function setUname($uname)
	{
		$this->uname = $uname;
	}

	public function setSource($source)
	{
		$this->source = $source;
	}

	public function setSourceSub($sourceSub)
	{
		$this->sourceSub = $sourceSub;
	}

	public function setLoginTime($loginTime)
	{
		$this->loginTime = $loginTime;
	}

}

File: pytools/examples/lua/client/protocols/ReqRoleCreate.lua
module("ReqRoleCreate", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqRoleCreate})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uid = 0
	self.uuid = 0
	self.sid = 0
	self.cid = 0
	self.os = 0
	self.version = 0
	self.uname = 0
	self.source = 0
	self.source_sub = 0
	self.login_time = 0
end


function encode(self)
	pack = packet.new()
	pack:writeU32(self.uid)
	pack:writeU32(self.uuid)
	pack:writeU16(self.sid)
	pack:writeU16(self.cid)
	pack:writeString(self.os)
	pack:writeString(self.version)
	pack:writeString(self.uname)
	pack:writeString(self.source)
	pack:writeString(self.source_sub)
	pack:writeU32(self.login_time)
	return pack:encode(Msg.P_REQ_ROLE_CREATE)
end


function setUid(self, uid)
	self.uid = uid
end

function setUuid(self, uuid)
	self.uuid = uuid
end

function setSid(self, sid)
	self.sid = sid
end

function setCid(self, cid)
	self.cid = cid
end

function setOs(self, os)
	self.os = os
end

function setVersion(self, version)
	self.version = version
end

function setUname(self, uname)
	self.uname = uname
end

function setSource(self, source)
	self.source = source
end

function setSourceSub(self, source_sub)
	self.source_sub = source_sub
end

function setLoginTime(self, login_time)
	self.login_time = login_time
end

